index.js

```diff
@@ -1,34 +1,53 @@
-import makeTemplate from "lodash.template";
-import { visitParents as visit } from "unist-util-visit-parents";
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @package rehype-lodash-template
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ * @copyright Viktor Yakubiv & Contributors
+ * @license MIT
+ */
 
-const attach = (options = {}) => {
-  const { values = {}, templateSettings } = options;
+import makeTemplate from "lodash.template";
+import { visitParents } from "unist-util-visit-parents";
 
-  const cache = new Map();
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ *
+ * @param {object} param0 { values, templateSettings }
+ * @returns Rehype Plugin
+ */
+export default function attach({ values, templateSettings }) {
+  const valueCache = new Map();
 
-  const template = (str) => {
-    if (!cache.has(str)) {
-      cache.set(str, makeTemplate(str, templateSettings));
+  const getTemplate = (identifier) => {
+    if (!valueCache.has(identifier)) {
+      valueCache.set(identifier, makeTemplate(identifier, templateSettings));
     }
 
-    return cache.get(str);
+    return valueCache.get(identifier);
   };
 
-  const substitute = (str) => {
-    const newString = template(str)(values);
+  const substitute = (identifier) => {
```

Comment:
It does not substitute an _identifier_, it actually operates on a _string_, but the string is called `str` to be shorted and to not ever conflict with TypeScript.